src/snowflake/cli/plugins/nativeapp/codegen/compiler.py

```diff
@@ -54,19 +55,28 @@ def compile_artifacts(self):
         Go through every artifact object in the project definition of a native app, and execute processors in order of specification for each of the artifact object.
         May have side-effects on the filesystem by either directly editing source files or the deploy root.
         """
+        should_proceed = False
         for artifact in self.artifacts:
-            for processor in artifact.processors:
-                artifact_processor = self._try_create_processor(
-                    processor_mapping=processor,
-                )
-                if artifact_processor is None:
-                    raise UnsupportedArtifactProcessorError(
-                        processor_name=processor.name
-                    )
-                else:
-                    artifact_processor.process(
-                        artifact_to_process=artifact, processor_mapping=processor
+            if artifact.processors:
```

Comment:
Do we need the extra `for` loop? I understand that this is not really a long list of artifacts, but wondering if we can combine it with something like this:

```
for artifact in self.artifacts:
     if not artifact.processors:
          continue
     with cc.phase("Invoking artifact processors"
     for processor in artifact.processors:
          ....
```

Reply:
If we have a start/end API we could use instead of the context manager API, it would be easy. Sadly, we do not.